domain/wollemi/service_format_test.go

```diff
@@ -226,6 +226,97 @@ func (t *ServiceSuite) TestService_GoFormat() {
 
 		require.NoError(t, wollemi.GoFormat(rewrite, []string{"app/server"}))
 	})
+
+	t.It("can resolve dependencies by import_path", func(t *T) {
+		data := t.GoFormatTestData()
+
+		delete(data.Stat, "third_party/go/github.com/spf13/BUILD.plz")
+		delete(data.Lstat, "third_party/go/github.com/spf13/BUILD.plz")
+		delete(data.Parse, "third_party/go/github.com/spf13/BUILD.plz")
+
+		data.Stat["third_party/go/github.com/spf13/cobra/BUILD.plz"] = &FileInfo{
+			FileName: "BUILD.plz",
+			FileMode: os.FileMode(420),
+		}
+		data.Lstat["third_party/go/github.com/spf13/cobra/BUILD.plz"] = data.Stat["third_party/go/github.com/spf13/cobra/BUILD.plz"]
+
+		data.Stat["third_party/go/github.com/spf13/pflag/BUILD.plz"] = &FileInfo{
+			FileName: "BUILD.plz",
+			FileMode: os.FileMode(420),
+		}
+		data.Lstat["third_party/go/github.com/spf13/pflag/BUILD.plz"] = data.Stat["third_party/go/github.com/spf13/pflag/BUILD.plz"]
+
+		data.Walk = make([]string, 0, len(data.Walk))
+		for path, _ := range data.Lstat {
+			data.Walk = append(data.Walk, path)
+		}
+
+		sort.Strings(data.Walk)
+
+		data.Parse["third_party/go/github.com/spf13/cobra/BUILD.plz"] = &please.BuildFile{
+			Path: "third_party/go/github.com/spf13/cobra/BUILD.plz",
+			Stmt: []please.Expr{
+				please.NewCallExpr("go_library", []please.Expr{
+					please.NewAssignExpr("=", "name", "cobra"),
+					please.NewAssignExpr("=", "srcs", please.NewGlob([]string{"*.go"}, "*_test.go")),
+					please.NewAssignExpr("=", "import_path", "github.com/spf13/cobra"),
+					please.NewAssignExpr("=", "deps", []string{"//third_party/go/github.com/spf13/pflag"}),
+					please.NewAssignExpr("=", "visibility", []string{"PUBLIC"}),
+				}),
+			},
+		}
+
+		data.Parse["third_party/go/github.com/spf13/pflag/BUILD.plz"] = &please.BuildFile{
+			Path: "third_party/go/github.com/spf13/pflag/BUILD.plz",
+			Stmt: []please.Expr{
+				please.NewCallExpr("go_library", []please.Expr{
+					please.NewAssignExpr("=", "name", "pflag"),
+					please.NewAssignExpr("=", "srcs", please.NewGlob([]string{"*.go"}, "*_test.go")),
+					please.NewAssignExpr("=", "import_path", "github.com/spf13/pflag"),
+					please.NewAssignExpr("=", "visibility", []string{"PUBLIC"}),
+				}),
+			},
+		}
```

Comment:
In the context of this test and the testdata being used the only thing which depends on pflag is cobra. Since the test is running wollemi on one specific build file, in this case app, we don't actually need the pflag rule to be defined. So we can just delete everything associated with pflag which should help slim down this test.